ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Block/SupportsDiffusionPipelining.hs

```diff
@@ -0,0 +1,273 @@
+{-# LANGUAGE DeriveAnyClass       #-}
+{-# LANGUAGE DeriveGeneric        #-}
+{-# LANGUAGE DeriveTraversable    #-}
+{-# LANGUAGE DerivingStrategies   #-}
+{-# LANGUAGE FlexibleContexts     #-}
+{-# LANGUAGE ScopedTypeVariables  #-}
+{-# LANGUAGE StandaloneDeriving   #-}
+{-# LANGUAGE TypeApplications     #-}
+{-# LANGUAGE TypeFamilies         #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+-- | See 'BlockSupportsDiffusionPipelining'.
+module Ouroboros.Consensus.Block.SupportsDiffusionPipelining (
+    BlockSupportsDiffusionPipelining (..)
+  , updateTentativeHeaderState
+    -- * @DerivingVia@ helpers
+    -- ** 'DisableDiffusionPipelining'
+  , DisableDiffusionPipelining (..)
+    -- ** 'SelectViewDiffusionPipelining'
+  , SelectViewDiffusionPipelining (..)
+  , SelectViewTentativeState (..)
+    -- * Data family instances
+  , BlockConfig (..)
+  , Header (..)
+  ) where
+
+import           Control.Monad (guard)
+import           Data.Coerce
+import           Data.Kind
+import           Data.Proxy
+import           GHC.Generics (Generic)
+import           NoThunks.Class
+import           Ouroboros.Consensus.Block.Abstract
+import           Ouroboros.Consensus.Block.SupportsProtocol
+import           Ouroboros.Consensus.Protocol.Abstract
+
+-- | Block functionality required to support __Block Diffusion Pipelining via
+-- Delayed Validation__ (DPvDV).
+--
+-- == High-level context
+--
+-- With DPvDV, a node is, under certain conditions, already announcing a new
+-- block to its downstream peers /before/ it has fully validated the block body.
+-- Concretely, the node maintains a /tentative header/ that, if present, extends
+-- the current selection, and is announced via the ChainSync servers to
+-- downstream peers.
+--
+-- Ideally, the body turns out to be valid, in which case the tentative header
```

Comment:
> the tentative header is cleared

I stumbled when reading this sentence. What exactly does "cleared" mean? I think there are two possible meanings, which both seem relevant: "cleared" as in "survived scrutiny" (like an IRL background check) versus "cleared" as in "reset" (like a bit in a byte). Maybe "set to `Nothing`" would be less ambiguous.

Reply:
Changed to "set to `Nothing`" :+1: